pyethapp/jsonrpc.py

```diff
@@ -928,9 +930,11 @@ def getBlockByNumber(self, block_id, include_transactions):
     def getTransactionByHash(self, tx_hash):
         try:
             tx, block, index = self.chain.chain.index.get_transaction(tx_hash)
+            if self.chain.chain.in_main_branch(block):
```

Comment:
@konradkonrad the RPC tests expect that the getTransactionByHash returns an empty result for the blocks which are not on the main branch, in other words "reverted" blocks.